rcl/test/rcl/test_time.cpp

```diff
@@ -193,14 +193,14 @@ TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), default_clock_instanciation) {
   EXPECT_EQ(retval, RCL_RET_OK) << rcl_get_error_string_safe();
   ASSERT_TRUE(rcl_clock_valid(&ros_clock));
 
-  rcl_clock_t * steady_clock =
-    reinterpret_cast<rcl_clock_t *>(calloc(1, sizeof(rcl_clock_t)));
+  rcl_clock_t * steady_clock = reinterpret_cast<rcl_clock_t *>(
+    allocator.zero_allocate(1, sizeof(rcl_clock_t), allocator.state));
```

Comment:
Is there no way around `reinterpret_cast` here? MISRA C++ guide on this is: Approval required. Rationale: Reinterpreting bits into a different type should not be necessary in a high level programming language. Exceptions could be made when a wrapper to a third party library working with opaque data types needs to be created.

Also SEI Cert C++ is not fond of it: https://wiki.sei.cmu.edu/confluence/display/cplusplus/VOID+EXP13-CPP.+Prefer+dynamic_cast+over+static_cast+over+reinterpret_cast.

Reply:
No, I believe it is required and I don't know any other way to do this when calling a C function. Malloc returns `void *` which must be cast to the pointer type you're using it as.

Reply:
For example they use the C-style cast in the example:

http://en.cppreference.com/w/cpp/memory/c/malloc

But we have a rule to not use the C-style cast when we can use `reinterpret_cast` or `static_cast` instead.

Reply:
Casting from a `void*` should be done with s `static_cast`.